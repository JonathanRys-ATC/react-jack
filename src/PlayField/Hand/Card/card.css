.PlayField__Card__card-face {
    background-color: #666;
    background: url(./images/card-back.png);
    background-size: contain;
    width: 76px;
    height: 100px;
    margin: 8px;
    border: 1px solid black;
    border-radius: 3px;
}

/* Spades */

.PlayField__Card__card-face.SA {
    background-color: #777;
    background: url(./images/SA.png);
}

.PlayField__Card__card-face.S2 {
    background-color: #777;
    background-image: "./images/S2.png";
}

.PlayField__Card__card-face.S3 {
    background-color: #777;
    background-image: "./images/S3.png";
}

.PlayField__Card__card-face.S4 {
    background-color: #777;
    background-image: "./images/S4.png";
}

.PlayField__Card__card-face.S5 {
    background-color: #777;
    background-image: "./images/S5.png";
}

.PlayField__Card__card-face.S6 {
    background-color: #777;
    background-image: "./images/S6.png";
}

.PlayField__Card__card-face.S7 {
    background-color: #777;
    background-image: "./images/S7.png";
}

.PlayField__Card__card-face.S8 {
    background-color: #777;
    background-image: "./images/S8.png";
}

.PlayField__Card__card-face.S9 {
    background-color: #777;
    background-image: "./images/S9.png";
}

.PlayField__Card__card-face.S10 {
    background-color: #777;
    background-image: "./images/S10.png";
}

.PlayField__Card__card-face.SJ {
    background-color: #777;
    background-image: "./images/SJ.png";
}

.PlayField__Card__card-face.SQ {
    background-color: #777;
    background-image: "./images/SQ.png";
}

.PlayField__Card__card-face.SK {
    background-color: #777;
    background-image: "./images/SK.png";
}

/* Hearts */

.PlayField__Card__card-face.HA {
    background-color: #777;
    background-image: "./images/HA.png";
}

.PlayField__Card__card-face.H2 {
    background-color: #777;
    background-image: "./images/H2.png";
}

.PlayField__Card__card-face.H3 {
    background-color: #777;
    background-image: "./images/H3.png";
}

.PlayField__Card__card-face.H4 {
    background-color: #777;
    background-image: "./images/H4.png";
}

.PlayField__Card__card-face.H5 {
    background-color: #777;
    background-image: "./images/H5.png";
}

.PlayField__Card__card-face.H6 {
    background-color: #777;
    background-image: "./images/H6.png";
}

.PlayField__Card__card-face.H7 {
    background-color: #777;
    background-image: "./images/H7.png";
}

.PlayField__Card__card-face.H8 {
    background-color: #777;
    background-image: "./images/H8.png";
}

.PlayField__Card__card-face.H9 {
    background-color: #777;
    background-image: "./images/H9.png";
}

.PlayField__Card__card-face.H10 {
    background-color: #777;
    background-image: "./images/H10.png";
}

.PlayField__Card__card-face.HJ {
    background-color: #777;
    background-image: "./images/HJ.png";
}

.PlayField__Card__card-face.HQ {
    background-color: #777;
    background-image: "./images/HQ.png";
}

.PlayField__Card__card-face.HK {
    background-color: #777;
    background-image: "./images/HK.png";
}

/* Diamonds */

.PlayField__Card__card-face.DA {
    background-color: #777;
    background-image: "./images/DA.png";
}

.PlayField__Card__card-face.D2 {
    background-color: #777;
    background-image: "./images/D2.png";
}

.PlayField__Card__card-face.D3 {
    background-color: #777;
    background-image: "./images/D3.png";
}

.PlayField__Card__card-face.D4 {
    background-color: #777;
    background-image: "./images/D4.png";
}

.PlayField__Card__card-face.D5 {
    background-color: #777;
    background-image: "./images/D5.png";
}

.PlayField__Card__card-face.D6 {
    background-color: #777;
    background-image: "./images/D6.png";
}

.PlayField__Card__card-face.D7 {
    background-color: #777;
    background-image: "./images/D7.png";
}

.PlayField__Card__card-face.D8 {
    background-color: #777;
    background-image: "./images/D8.png";
}

.PlayField__Card__card-face.D9 {
    background-color: #777;
    background-image: "./images/D9.png";
}

.PlayField__Card__card-face.D10 {
    background-color: #777;
    background-image: "./images/D10.png";
}

.PlayField__Card__card-face.DJ {
    background-color: #777;
    background-image: "./images/DJ.png";
}

.PlayField__Card__card-face.DQ {
    background-color: #777;
    background-image: "./images/DQ.png";
}

.PlayField__Card__card-face.DK {
    background-color: #777;
    background-image: "./images/DK.png";
}

/* Clubs */

.PlayField__Card__card-face.CA {
    background-color: #777;
    background-image: "./images/CA.png";
}

.PlayField__Card__card-face.C2 {
    background-color: #777;
    background-image: "./images/C2.png";
}

.PlayField__Card__card-face.C3 {
    background-color: #777;
    background-image: "./images/C3.png";
}

.PlayField__Card__card-face.C4 {
    background-color: #777;
    background-image: "./images/C4.png";
}

.PlayField__Card__card-face.C5 {
    background-color: #777;
    background-image: "./images/C5.png";
}

.PlayField__Card__card-face.C6 {
    background-color: #777;
    background-image: "./images/C6.png";
}

.PlayField__Card__card-face.C7 {
    background-color: #777;
    background-image: "./images/C7.png";
}

.PlayField__Card__card-face.C8 {
    background-color: #777;
    background-image: "./images/C8.png";
}

.PlayField__Card__card-face.C9 {
    background-color: #777;
    background-image: "./images/C9.png";
}

.PlayField__Card__card-face.C10 {
    background-color: #777;
    background-image: "./images/C10.png";
}

.PlayField__Card__card-face.CJ {
    background-color: #777;
    background-image: "./images/CJ.png";
}

.PlayField__Card__card-face.CQ {
    background-color: #777;
    background-image: "./images/CQ.png";
}

.PlayField__Card__card-face.CK {
    background-color: #777;
    background-image: "./images/CK.png";
}