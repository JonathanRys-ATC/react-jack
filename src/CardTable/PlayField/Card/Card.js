import React, { Component } from 'react'
import PropTypes from 'prop-types';

import "./card.css"

export default class Card extends Component {
    shouldComponentUpdate(nextProps) {
        if (this.props.faceUp === nextProps.faceUp &&
            this.props.card === nextProps.card) return false;
        return true;
    }

    render() {
        return (
            <div className={`PlayField__Card__card-face ${this.props.faceUp ? this.props.card : ""}`}></div>
        )
    }
}

Card.propTypes = {
    card: PropTypes.string.isRequired,
    faceUp: PropTypes.number.isRequired,
}