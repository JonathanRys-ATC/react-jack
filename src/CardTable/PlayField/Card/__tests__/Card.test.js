import React from 'react'
import { shallow } from 'enzyme'

import Card from '../Card'

const mockProps = { card: "SA", faceUp: false }

describe("Test Card component", () => {
    it("should display the correct card", () => {
        let component = shallow(<Card {...mockProps} />)
        expect(component).toMatchSnapshot()

        mockProps.faceup = true

        component = shallow(<Card {...mockProps} />)
        expect(component).toMatchSnapshot()
    })
})
