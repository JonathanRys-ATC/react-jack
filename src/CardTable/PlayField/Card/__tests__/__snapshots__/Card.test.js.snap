// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Card component should display the correct card 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    card="SA"
    faceUp={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "SA",
      "className": "PlayField__Card__card-face ",
    },
    "ref": null,
    "rendered": "SA",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "SA",
        "className": "PlayField__Card__card-face ",
      },
      "ref": null,
      "rendered": "SA",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Test Card component should display the correct card 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    card="SA"
    faceUp={1}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "SA",
      "className": "PlayField__Card__card-face SA",
    },
    "ref": null,
    "rendered": "SA",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "SA",
        "className": "PlayField__Card__card-face SA",
      },
      "ref": null,
      "rendered": "SA",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Test Card component should display the correct card 3`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    card="SA"
    faceUp={1}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "SA",
      "className": "PlayField__Card__card-face SA",
    },
    "ref": null,
    "rendered": "SA",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "SA",
        "className": "PlayField__Card__card-face SA",
      },
      "ref": null,
      "rendered": "SA",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
